src/main/java/com/google/devtools/build/lib/analysis/configuredtargets/OutputFileConfiguredTarget.java

```diff
@@ -114,6 +115,14 @@ public boolean hasOutputLicenses() {
         .hasOutputLicenses();
   }
 
+  @Nullable
+  @Override
+  public <P extends TransitiveInfoProvider> P getProvider(Class<P> providerClass) {
```

Comment:
What's this for?

Reply:
It's the only way that I could get the check for incompatibility on dependencies to work.

I.e. this snippet in `ConfiguredTargetFactory.java` :
```java
      // This is incompatible if one of the dependencies is as well.
      for (ConfiguredTargetAndData infoCollection : prerequisiteMap.values()) {
        if (infoCollection.getConfiguredTarget().getProvider(IncompatiblePlatformProvider.class) != null) {
          isIncompatible = true;
        }
      }
```

In other words, I need to check the provider of the generating rule. Otherwise the artifacts themselves will never have provider set on their own. Unless I somehow pass the provider to the generated artifacts also? Haven't looked into that. This approach seemed to make more sense at the time.

Reply:
I think I'm confused about the existence of this method in the base class: https://github.com/bazelbuild/bazel/blob/773b95f947fbba51185db6548ba63c2008bed8ec/src/main/java/com/google/devtools/build/lib/analysis/configuredtargets/FileConfiguredTarget.java#L110

That suggests to me other Bazel code could already be calling `ConfiguredTarget.getProvider(Class<P> providerClass)` which currently returns the above linked logic. If your change overrides that to provide something different, could that break other invariants?

Unless I'm getting my logic mixed up, maybe it's safer for your code to special-case check`is instance of OutputFileConfiguredTarget` and call `((OutputFileConfiguredTarget) configuredTarget.getGeneratingRule().getProvider()` instead.

It's not ideally clean. But as far as I can see it's safer.